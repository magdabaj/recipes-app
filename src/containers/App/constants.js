export const LOAD_RECIPES = 'app/App/LOAD_RECIPES';
export const LOAD_RECIPES_SUCCESS = 'app/App/LOAD_RECIPES_SUCCESS';
export const LOAD_RECIPES_ERROR = 'app/App/LOAD_RECIPES_ERROR';

export const LOAD_TAGS = 'app/App/LOAD_TAGS';
export const LOAD_TAGS_SUCCESS = 'app/App/LOAD_TAGS_SUCCESS';
export const LOAD_TAGS_ERROR = 'app/App/LOAD_TAGS_ERROR';