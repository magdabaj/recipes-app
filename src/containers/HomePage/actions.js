/*
 *
 * RecipesHomePage actions
 *
 */

import {
    CHANGE_PAGE,
    CHANGE_PAGE_ERROR,
    CHANGE_PAGE_SUCCESS,
    LOAD_RECIPES,
    LOAD_RECIPES_BY_TAG,
    LOAD_RECIPES_BY_TAG_ERROR,
    LOAD_RECIPES_BY_TAG_SUCCESS,
    LOAD_RECIPES_ERROR,
    LOAD_RECIPES_SUCCESS,
    LOAD_TAGS,
    LOAD_TAGS_ERROR,
    LOAD_TAGS_SUCCESS,
} from './constants';

export const loadRecipes = page => ({
    type: LOAD_RECIPES,
    page,
});

export const loadRecipesSuccess = response => ({
    type: LOAD_RECIPES_SUCCESS,
    response,
});

export const loadRecipesError = error => ({
    type: LOAD_RECIPES_ERROR,
    error,
});

export const loadRecipesByTag = (tagId, page) => ({
    type: LOAD_RECIPES_BY_TAG,
    tagId,
    page,
});

export const loadRecipesByTagSuccess = response => ({
    type: LOAD_RECIPES_BY_TAG_SUCCESS,
    response,
});

export const loadRecipesByTagError = error => ({
    type: LOAD_RECIPES_BY_TAG_ERROR,
    error,
});

// export const changePage = page => ({
//   type: CHANGE_PAGE,
//   page,
// });
//
// export const changePageSuccess = response => ({
//   type: CHANGE_PAGE_SUCCESS,
//   response,
// });
//
// export const changePageError = error => ({
//   type: CHANGE_PAGE_ERROR,
//   error,
// })
//
