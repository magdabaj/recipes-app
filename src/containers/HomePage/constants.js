/*
 *
 * RecipesHomePage constants
 *
 */

export const LOAD_RECIPES = 'app/RecipesHomePage/LOAD_RECIPES';
export const LOAD_RECIPES_SUCCESS = 'app/RecipesHomePage/LOAD_RECIPES_SUCCESS';
export const LOAD_RECIPES_ERROR = 'app/RecipesHomePage/LOAD_RECIPES_ERROR';

export const SEND_RATING = 'app/RecipesHomePage/SEND_RATING';
export const SEND_RATING_SUCCESS = 'app/RecipesHomePage/SEND_RATING_SUCCESS';
export const SEND_RATING_ERROR = 'app/RecipesHomePage/SEND_RATING_ERROR';

export const LOAD_RECIPES_BY_TAG = 'app/RecipesHomePage/LOAD_RECIPES_BY_TAG';
export const LOAD_RECIPES_BY_TAG_SUCCESS = 'app/RecipesHomePage/LOAD_RECIPES_BY_TAG_SUCCESS';
export const LOAD_RECIPES_BY_TAG_ERROR = 'app/RecipesHomePage/LOAD_RECIPES_BY_TAG_ERROR';

// export const CHANGE_PAGE = "app/RecipesHomePage/CHANGE_PAGE";
// export const CHANGE_PAGE_SUCCESS = "app/RecipesHomePage/CHANGE_PAGE_SUCCESS";
// export const CHANGE_PAGE_ERROR = "app/RecipesHomePage/CHANGE_PAGE_ERROR";
