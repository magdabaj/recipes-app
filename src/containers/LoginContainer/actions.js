/*
 *
 * LoginFormContainer actions
 *
 */

import {
    AUTHENTICATE,
    AUTHENTICATE_ERROR,
    AUTHENTICATE_SUCCESS, SET_ERROR,
    SIGN_IN,
    SIGN_IN_ERROR,
    SIGN_IN_SUCCESS,
    SIGN_OUT,
    SIGN_OUT_ERROR,
    SIGN_OUT_SUCCESS,
    SIGN_UP_USER,
    SIGN_UP_USER_ERROR,
    SIGN_UP_USER_SUCCESS
} from './constants'

export const setError = () => ({
    type: SET_ERROR
})

export const signUp = ({ email, password }) => ({
    type: SIGN_UP_USER,
    email,
    password,
});

export const signUpSuccess = (json, email) => ({
    type: SIGN_UP_USER_SUCCESS,
    ...json,
    email,
});

export const signUpError = error => ({
    type: SIGN_UP_USER_ERROR,
    error,
});

export const signOut = () => ({
    type: SIGN_OUT,
});

export const signOutError = error => ({
    type: SIGN_OUT_ERROR,
    error,
});

export const signOutSuccess = json => ({
    type: SIGN_OUT_SUCCESS,
    ...json,
});

export const signIn = ({ email, password }) => ({
    type: SIGN_IN,
    email,
    password,
});

export const signInSuccess = (json, email) => ({
    type: SIGN_IN_SUCCESS,
    ...json,
    email,
});

export const signInError = error => ({
    type: SIGN_IN_ERROR,
    error,
});

export const authenticate = () => ({
    type: AUTHENTICATE,
});

export const authenticateSuccess = json => ({
    type: AUTHENTICATE_SUCCESS,
    ...json,
});

export const authenticateError = error => ({
    type: AUTHENTICATE_ERROR,
    error,
});
