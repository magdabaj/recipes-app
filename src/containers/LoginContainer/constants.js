/*
 *
 * LoginFormContainer constants
 *
 */

export const SIGN_UP_USER = 'app/LoginFormContainer/SIGN_UP_USER';
export const SIGN_UP_USER_SUCCESS = 'app/LoginFormContainer/SIGN_UP_USER_SUCCESS';
export const SIGN_UP_USER_ERROR = 'app/LoginFormContainer/SIGN_UP_USER_ERROR';

export const SIGN_IN = 'app/LoginFormContainer/SIGN_IN';
export const SIGN_IN_SUCCESS = 'app/LoginFormContainer/SIGN_IN_SUCCESS';
export const SIGN_IN_ERROR = 'app/LoginFormContainer/SIGN_IN_ERROR';

export const SIGN_OUT = 'app/LoginFormContainer/SIGN_OUT';
export const SIGN_OUT_SUCCESS = 'app/LoginFormContainer/SIGN_OUT_SUCCESS';
export const SIGN_OUT_ERROR = 'app/LoginFormContainer/SIGN_OUT_ERROR';

export const AUTHENTICATE = 'app/LoginFormContainer/AUTHENTICATE';
export const AUTHENTICATE_SUCCESS = 'app/LoginFormContainer/AUTHENTICATE_SUCCESS';
export const AUTHENTICATE_ERROR = 'app/LoginFormContainer/AUTHENTICATE_ERROR';

export const CLEAR_STATUS = 'app/LoginFormContainer/CLEAR_STATUS';
export const CLEAR_LOGOUT_STATUS = 'app/LoginFormContainer/CLEAR_LOGOUT_STATUS';
