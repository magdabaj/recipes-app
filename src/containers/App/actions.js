import {LOAD_RECIPES_ERROR, LOAD_TAGS, LOAD_TAGS_ERROR, LOAD_TAGS_SUCCESS} from "./constants";

// export const loadRecipes = page => ({
//     type: LOAD_RECIPES,
//     page,
// });
//
// export const loadRecipesSuccess = response => ({
//     type: LOAD_RECIPES_SUCCESS,
//     response,
// });

export const loadRecipesError = error => ({
    type: LOAD_RECIPES_ERROR,
    error,
});

export const loadTags = () => ({
    type: LOAD_TAGS,
});

export const loadTagsSuccess = tags => ({
    type: LOAD_TAGS_SUCCESS,
    tags,
});

export const loadTagsError = error => ({
    type: LOAD_TAGS_ERROR,
    error,
});