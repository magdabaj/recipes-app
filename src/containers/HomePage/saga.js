import { take, call, put, select, takeLatest, all } from 'redux-saga/effects';
import {
    loadRecipesByTagError,
    loadRecipesByTagSuccess,
    loadRecipesError,
    loadRecipesSuccess,
} from './actions';
import { LOAD_RECIPES, LOAD_RECIPES_BY_TAG } from './constants';
import { fetchAllRecipes, fetchRecipesByTag } from '../../utils/api/recipe';

export function* loadRecipesSaga(action) {
    const { page } = action;
    try {
        const response = yield call(fetchAllRecipes, page);
        yield put(loadRecipesSuccess(response));
    } catch (error) {
        yield put(loadRecipesError(error.message));
    }
}

export function* watchRecipesLoad() {
    yield takeLatest(LOAD_RECIPES, loadRecipesSaga);
}

export function* loadRecipesByTagSaga(action) {
    const { tagId, page } = action;

    try {
        const response = yield call(fetchRecipesByTag, tagId, page);
        yield put(loadRecipesByTagSuccess(response));
    } catch (e) {
        yield put(loadRecipesByTagError(e.message));
    }
}

export function* watchRecipesByTagLoad() {
    yield takeLatest(LOAD_RECIPES_BY_TAG, loadRecipesByTagSaga);
}