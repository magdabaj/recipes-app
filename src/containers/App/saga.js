import {call, put, takeLatest} from "redux-saga/effects";
import {fetchAllTags} from "../../utils/api/tags";
import {loadTagsError, loadTagsSuccess} from "./actions";
import {LOAD_TAGS} from "./constants";

export function* loadTagsSaga() {
    try {
        const tags = yield call(fetchAllTags);
        yield put(loadTagsSuccess(tags));
    } catch (error) {
        yield put(loadTagsError(error.message));
    }
}

export function* watchTagsLoad() {
    yield takeLatest(LOAD_TAGS, loadTagsSaga);
}