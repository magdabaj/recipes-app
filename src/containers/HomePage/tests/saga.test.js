import {loadRecipesSaga} from "../saga";
import { put } from 'redux-saga/effects'
import {expect} from "@jest/globals";
// import 'core-js/stable';
// import 'regenerator-runtime/runtime'
import {fakeRecipes} from "../../../utils/testHelpers/fixtures/recipes";
import {loadRecipesSuccess} from "../actions";
import "@babel/polyfill/noConflict"
import {fakeNextPage} from "../../../utils/testHelpers/fixtures/page";
import {LOAD_RECIPES} from "../constants";

describe('HomePage saga', () => {
    let loadRecipesSagaGenerator;

    beforeEach(() => {
        loadRecipesSagaGenerator = loadRecipesSaga({type: LOAD_RECIPES,page: 1})

        const callDescriptor = loadRecipesSagaGenerator.next().value
        expect(callDescriptor).toMatchSnapshot()
        // todo how shall I test this???
        // const callPageDescriptor = await waitFor(() => loadRecipesSagaGenerator.next({ page: 1 }).value);
        // await waitForExpect(() => expect(callPageDescriptor).toMatchSnapshot())
    })

    it('should dispatch the loadRecipesSuccess action if it gets the data successfully', function () {
        const response = {
            items: fakeRecipes,
            nextPage:fakeNextPage,
            totalPages: 5
        }
        const putLoadRecipesSuccess = loadRecipesSagaGenerator.next(response).value;
        expect(putLoadRecipesSuccess).toEqual(put(loadRecipesSuccess(response)))
    });
})