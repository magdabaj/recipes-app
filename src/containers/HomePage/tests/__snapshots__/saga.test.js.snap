// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage saga should dispatch the loadRecipesSuccess action if it gets the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      1,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
